### PR TITLE
Claim investment

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR connects the claiming investment to the investment flow.

 # To Test

1. You can use accounts:
- 0xdB45e7913596C9dcd8Dca6D16F6616d4EbF647f1
- 0xd72bfEceAA089601a7B14d3cc0A79c56553266eA
- 0xD4612f0C63276B6859eB8030bfB80C1e677d0de9
- 0xD3E09Fe5d2556443e4FEBd9bBc3DCaC0e54606E3
- 0xd30544A41E10085093A65F8B7557b4ABbe0b3554

 
2. The try to invest 100% of the GNO (you will need to approve first)

### CODE REVIEW

File: src/custom/pages/Claim/index.tsx [hunk 1]
```diff
@@ -146,11 +141,28 @@ export default function Claim() {
...
+    // check if there are any selected (paid) claims
+    let inputData
+    if (!selected.length) {
+      inputData = freeClaims.map(({ index }) => ({ index }))
+      console.log('Starting claiming with', inputData)
+      sendTransaction(inputData)
+    } else if (investFlowStep == 2) {
+      // Free claimings + selected investment oportunities
+      const selectedIndex = [...getIndexes(freeClaims), ...selected]
+      inputData = selectedIndex.reduce<EnhancedUserClaimData[]>((acc, idx: number) => {
```
Comment (anxolin) [inline src/custom/pages/Claim/index.tsx:155]: i changed this into a reduce instead of a map + filter, because TS was giving me a hard time. It was thinking the array of clamings could contain undefined when i was filtering them...
reduce worked :)

Comment (W3stside) [reply src/custom/pages/Claim/index.tsx:155]: should we break apart this fn?

Comment (anxolin) [reply src/custom/pages/Claim/index.tsx:155]: u mean the reduce? we could, but not sure it would add a lot of value. 
if we had time, would be nice to cleanup things a bit, cause the logic still is a bit hard to follow

**Comment by elena-zh** [comment]
Could you please provide a preview link?

**Review by alfetopito — APPROVED** [review]
Worked great!

Tested the ETH claim flow https://rinkeby.etherscan.io/tx/0x6e04ae0a69d91f1888246554aec7d478ff14caa7c7ac252092f42eedcc4342aa

File: src/custom/pages/Claim/index.tsx [hunk 2]
```diff
@@ -146,11 +141,28 @@ export default function Claim() {
...
+        const claim = userClaimData.find(({ index }) => idx === index)
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/custom/pages/Claim/index.tsx:156]: Yep, will add the TODO. @nenadV91 is working in that part, we will need to read his state and change those amounts
